vm/devices/get/guest_emulation_device/src/lib.rs

```diff
@@ -179,12 +179,23 @@ pub struct GuestEmulationDevice {
     #[inspect(skip)]
     waiting_for_vtl0_start: Vec<mesh::OneshotSender<Result<(), Vtl0StartError>>>,
 
-    #[inspect(skip)]
-    vmgs: Vec<u8>,
+    vmgs: Option<VmgsState>,
+
     #[inspect(with = "Option::is_some")]
     save_restore_buf: Option<Vec<u8>>,
 }
 
+#[derive(Inspect)]
+struct VmgsState {
+    /// The underlying VMGS disk.
+    disk: Arc<dyn SimpleDisk>,
+    /// Memory for the guest to DMA to/from.
```

Comment:
and on a related note: is there any way to leverage the fact that this memory is not actually continuously "shared" in order to avoid maintaining the separate `buf: Vec<u8>` below?

Reply:
Ehh, it's not worth it to save 8KB.

Reply:
In this use case? Yeah, fair enough.
Maybe we'll find a more compelling use case for this sort of API in the future, and we revisit it later.

Reply:
Yeah. I think we can have something like `GuestMemory::try_bytes_mut(&mut self) -> Option<&mut [u8]>`. But it will only work for memories allocated via `GuestMemory::allocate`, and only if you haven't cloned your `GuestMemory`. So I'm a little worried about the confusion it might cause.

Reply:
(What might be more interesting is to think a little harder about how DMA buffers are specified, and in particular, to decouple them from `GuestMemory` altogether. Not sure yet.)

Reply:
> (What might be more interesting is to think a little harder about how DMA buffers are specified, and in particular, to decouple them from `GuestMemory` altogether. Not sure yet.)

Indeed, that was going to be my follow-up thought. It seems like a bit of indirection here could be useful, given that we'll keep finding places to use these sorts of Disk APIs that aren't in the presence of the VMM itself. Today, that's things like vmgstool, etc..., but I can forsee a future where we might spin up a home-grown version of `qemu-image convert` where we'd want to do disk-ey stuff in another standalone tool.